syntax = "proto3";

package user;

service UserService {
    rpc RegisterEmail(RegisterEmailRequest) returns (RegisterEmailResponse);
    rpc RegisterMentor(RegisterMentorRequest) returns (RegisterMentorResponse);
    rpc LoginEmail(LoginEmailRequest) returns (LoginEmailResponse);
    rpc RefreshToken(RefreshTokenRequest) returns (RefreshTokenResponse);
    rpc AuthGoogle(AuthGoogleRequest) returns (AuthGoogleResponse);
    rpc RequestPasswordReset (RequestPasswordResetRequest)
      returns (RequestPasswordResetResponse);
    rpc ConfirmPasswordReset (ConfirmPasswordResetRequest)
      returns (ConfirmPasswordResetResponse);
}

message RegisterEmailRequest {
    string email = 1;
    string password = 2;
    string phone_number = 3;
}

message RegisterEmailResponse {
    string user_id = 1;
    string access_token = 2;
    string refresh_token = 3;
}

message LoginEmailRequest {
    string email = 1;
    string password = 2;
}

message LoginEmailResponse {
    string user_id = 1;
    string access_token = 2;
    string refresh_token = 3;
}

message RefreshTokenRequest {
    string refresh_token = 1;
}

message RefreshTokenResponse {
    string access_token = 1;
    string refresh_token = 2;
}

message AuthGoogleRequest {
    string token = 1;
}

message AuthGoogleResponse {
    string user_id = 1;
    string access_token = 2;
    string refresh_token = 3;
}

message RequestPasswordResetRequest {
  string email = 1;
}

message RequestPasswordResetResponse {
  string status = 1;
}

message ConfirmPasswordResetRequest {
  string email = 1;
  string code = 2;
  string new_password = 3;
}

message ConfirmPasswordResetResponse {
  string status = 1;
}

message RegisterMentorRequest {
    string email = 1;
    string password = 2;
    string phone_number = 3;
    string role = 4;
}

message RegisterMentorResponse {
    bool success = 1;
}